bin/jobsub_fetchlog

```diff
@@ -51,6 +54,134 @@ from version import print_version, print_support_email
 
 verbose = 0
 
+# environment variable containing base fetchlog server url
+_FETCHLOG_URL_ENV = "JOBSUB_FETCHLOG_URL"
+# archive download chunk size in bytes
+_CHUNK_SIZE = 1024 * 1024
+
+
+def fetch_from_condor(
+    jobid: str, destdir: Optional[str], archive_format: str, partial: bool
+):
+    # find where the condor_transfer_data will put the output
+    j = Job(jobid)
+    iwd = j.get_attribute("SUBMIT_Iwd")
+    if verbose:
+        print(f"job output to {iwd}")
+    # make sure it exists, create if not
+    try:
+        os.stat(iwd)
+    except FileNotFoundError:
+        os.makedirs(iwd, mode=0o750)
+
+    # get the output sandbox
+    try:
+        j.transfer_data(partial)
+        transfer_complete = True
+    except htcondor.HTCondorIOError as e1:
+        print(f"Error in transfer_data(): {str(e1)}")
+    files = os.listdir(iwd)
+
+    if destdir is not None:
+        # If the user wants output in a specific directory, copy files there,
+        # don't build an archive. Old jobsub would get an archive from the
+        # server, upack it into the dest dir, then delete the archive.
+        owd = destdir
+        try:
+            os.stat(owd)
+        except FileNotFoundError:
+            os.makedirs(owd, mode=0o750)
+        for f in files:
+            shutil.copy2(os.path.join(iwd, f), owd)  # copy2 tries to preserve metadata
+    else:
+        # build archive
+        cmd = []
+        if archive_format == "tar":
+            cmd = ["/usr/bin/tar", "-C", iwd, "-czf", f"{str(j)}.tgz"] + files
+            # -C: move into directory so paths are relative
+            # -c: create
+            # -z: gzip
+            # -f: filename
+        elif archive_format == "zip":
+            cmd = ["/usr/bin/zip", "-jq", f"{str(j)}.zip"] + [
+                os.path.join(iwd, f) for f in files
+            ]
+            # -j: junk (don't record) directory names
+            # -q: quiet
+        else:
+            raise Exception(f'unknown archive format "{archive_format}"')
+        if verbose:
+            print(f'running "{cmd}"')
+        p = subprocess.Popen(cmd)
+        if p.wait() != 0:
+            raise Exception(f"error creating archive")
+
+    cleanup({"submitdir": iwd})
+
+    if not transfer_complete:
+        print("Transfer may be incomplete.")
+
+
+def fetch_from_landscape(
+    jobid: str, destdir: Optional[str], archive_format: str, partial: bool
+):
+    # landscape doesn't support zip, does anyone actually use it?
+    if archive_format != "tar":
+        raise Exception(f'unknown/unsupported archive format "{archive_format}"')
+
+    if _FETCHLOG_URL_ENV not in os.environ:
+        raise Exception(
+            f"{_FETCHLOG_URL_ENV} not set. You may still be able to download with --condor"
```

Comment:
Could you please add "in the environment" or something like that to the first sentence?  Just to make it crystal-clear to the user that they need an env. var `JOBSUB_FETCHLOG_URL`?

Reply:
LGTM.  Thanks!